x-pack/plugins/enterprise_search/server/routes/enterprise_search/crawler/crawler.ts

```diff
@@ -35,55 +35,75 @@ export function registerCrawlerRoutes(routeDependencies: RouteDependencies) {
-          ),
-          response,
-          statusCode: 409,
-        });
-      }
-      const crawler = await fetchCrawlerByIndexName(client, request.body.index_name);
-      if (crawler) {
-        return createError({
-          errorCode: ErrorCode.CRAWLER_ALREADY_EXISTS,
-          message: i18n.translate(
-            'xpack.enterpriseSearch.server.routes.addCrawler.crawlerExistsError',
-            {
-              defaultMessage: 'A crawler for this index already exists',
-            }
-          ),
-          response,
-          statusCode: 409,
+
+      try {
+        const indexExists = await client.asCurrentUser.indices.exists({
+          index: request.body.index_name,
         });
-      }
+        if (indexExists) {
+          return createError({
+            errorCode: ErrorCode.INDEX_ALREADY_EXISTS,
+            message: i18n.translate(
+              'xpack.enterpriseSearch.server.routes.addCrawler.indexExistsError',
+              {
+                defaultMessage: 'This index already exists',
+              }
+            ),
+            response,
+            statusCode: 409,
+          });
+        }
+        const crawler = await fetchCrawlerByIndexName(client, request.body.index_name);
+        if (crawler) {
+          return createError({
+            errorCode: ErrorCode.CRAWLER_ALREADY_EXISTS,
+            message: i18n.translate(
+              'xpack.enterpriseSearch.server.routes.addCrawler.crawlerExistsError',
+              {
+                defaultMessage: 'A crawler for this index already exists',
+              }
+            ),
+            response,
+            statusCode: 409,
+          });
+        }
 
-      const connector = await fetchConnectorByIndexName(client, request.body.index_name);
+        const connector = await fetchConnectorByIndexName(client, request.body.index_name);
 
-      if (connector) {
-        return createError({
-          errorCode: ErrorCode.CONNECTOR_DOCUMENT_ALREADY_EXISTS,
-          message: i18n.translate(
-            'xpack.enterpriseSearch.server.routes.addCrawler.connectorExistsError',
-            {
-              defaultMessage: 'A connector for this index already exists',
-            }
-          ),
-          response,
-          statusCode: 409,
-        });
+        if (connector) {
+          return createError({
+            errorCode: ErrorCode.CONNECTOR_DOCUMENT_ALREADY_EXISTS,
+            message: i18n.translate(
+              'xpack.enterpriseSearch.server.routes.addCrawler.connectorExistsError',
+              {
+                defaultMessage: 'A connector for this index already exists',
+              }
+            ),
+            response,
+            statusCode: 409,
+          });
+        }
+
+        return enterpriseSearchRequestHandler.createRequest({
+          path: '/api/ent/v1/internal/indices',
+        })(context, request, response);
+      } catch (e) {
+        // 403 UNAUTHORIZED
+        if (e.meta.statusCode === 403) {
```

Comment:
Can you add this as an isUnauthorizedError to this file: 'x-pack/plugins/enterprise_search/server/utils/identify_exceptions.ts'